### ISSUE TITLE
participantVersion and latest are not mutually exclusive in CanDeploy

### ISSUE THREAD

**Issue by halfzebra** [pact-foundation/pact-node#157]
Hello friends, I'm happy to use Pact!

I've noticed something in how `pact-node` parses the options to `can-i-deploy`.

## Software versions

-   **Library version**:
    - @pact-foundation/pact-node ^8.1.2

## Expected behavior

I think `pact can-i-deploy` command and `CanDeploy` should allow me to check whether I can deploy current service against the specified tag without specifying a version.

## Actual behavior

I'm using the Broker credentials from [Pact Workshop](https://github.com/DiUS/pact-workshop-js/blob/master/package.json) to illustrate this:

```bash
npx pact can-i-deploy
        --participant 'Our Provider' \
        --latest prod \
        --pact-broker https://test.pact.dius.com.au \
        --username dXfltyFMgNOFZAxr8io9wJ37iUpY42M \
        --password O5AIZWxelWbLvqMd8PkAVycBJh2Psyg1

Error: Missing option --participant-version.
```

I might be wrong, but I think this is incorrect behavior. 🤔 

These checks https://github.com/pact-foundation/pact-node/blob/master/src/can-deploy.ts#L50 should be checking if `latest` is specified and not require `participantVersion` in that case.

I can work on the PR to change this.

What do you think?

**Comment by bethesque** [comment]
Why is this cli even wrapped @mboudreau. What value is it providing?

**Comment by mboudreau** [comment]
@bethesque typescript typings, variable checking (spinning up a binary is slow), coding standards, and fixing issues through abstraction where possible.  For a while, we didn't even have the provider verifier as a binary so we implemented it ourselves as a separate thing.

Essentially, just trying to add some normalcy to how node work without having the ruby implementation details bleed through.  It definitely could use a major refactor if we're going to use the rust c library though.

**Comment by bethesque** [comment]
It's a CLI, what rubyisms are in it? What coding standards and abstractions are you concerned about?

**Comment by halfzebra** [comment]
I feel like it might be excessive to perform assertions on the user input which is not used directly in the JavaScript code.

If assertions are going to be removed, users who choose to run `CanDeploy` directly might experience difficulties figuring out what exactly is wrong with the options they are passing, because the binary does not take into account weird situations when someone passes an array as a username.

 I'd propose to fix this by not expecting `participantVersion` if seemingly valid `latest` is provided for now.

What do you think?

**Comment by mefellows** [comment]
As a related aside, see #161 (the can-i-deploy task is not available to be called outside of the Node API if that is convenient)

**Comment by TimothyJones** [comment]
#173 has a fix for this, with a much improved options interface.

**Comment by TimothyJones** [comment]
This is released in pact-node 10.0.0, which is a breaking change to the way options work. See [the changelog](https://github.com/pact-foundation/pact-node/blob/master/CHANGELOG.md) for more details

**Comment by halfzebra** [comment]
Thanks for the update!
